src/Mapster/Utils/ReflectionUtils.cs

```diff
@@ -37,6 +38,24 @@ public static IEnumerable<IMemberModel> GetPublicFieldsAndProperties(this Type t
             return properties.Concat(fields);
         }
 
+        public static IEnumerable<IMemberModel> GetPrivateFieldsAndProperties(this Type type, bool allowNoSetter = true)
+        {
+            var properties = type.GetProperties(BindingFlags.Instance | BindingFlags.NonPublic)
+                .Where(x => allowNoSetter || x.CanWrite)
+                .Select(CreateModel);
+
+            var fields = type.GetFields(BindingFlags.Instance | BindingFlags.NonPublic)
+                .Where(x => (allowNoSetter || !x.IsInitOnly) && !IsBackingField(x))
+                .Select(CreateModel);
+
+            return properties.Concat(fields);
+        }
```

Comment:
I think we should combined `GetPrivateFieldsAndProperties` and `GetPublicFieldsAndProperties` by sending another parameter `isNonPublic` in order to unify logic.